app/controllers/application_controller.rb

```diff
@@ -1140,32 +1143,94 @@ def listicon_image(view, id = nil)
              klass.find(id)    # Read the record from the db
            end
 
-    p  = "/images/icons/"
-    pn = "#{p}new/"
-
-    image = case item
-            when ExtManagementSystem   then "#{pn}/vendor-#{item.image_name}.png"
-            when Filesystem            then "#{p}ico/win/#{item.image_name.downcase}.ico"
-            when Host                  then "#{pn}vendor-#{item.vmm_vendor.downcase}.png"
-            when MiqEventDefinition    then "#{pn}event-#{item.name.downcase}.png"
-            when MiqRequest
-              pn + case item.request_status.to_s.downcase
-                   when "ok"    then "checkmark.png"
-                   when "error" then "x.png"
-                   else              "#{@listicon.downcase}.png"
-                   end
-            when RegistryItem          then "#{pn}#{item.image_name.downcase}.png"
-            when ResourcePool          then "#{pn}#{item.vapp ? "vapp" : "resource_pool"}.png"
-            when VmOrTemplate          then "#{pn}vendor-#{item.vendor.downcase}.png"
-            when ServiceResource       then "#{pn}#{item.resource_type.to_s == "VmOrTemplate" ? "vm" : "service_template"}.png"
-            when Storage               then "#{pn}piecharts/datastore/#{calculate_pct_img(item.v_free_space_percent_of_total)}.png"
-            when OsProcess, EventLog   then "#{pn}#{@listicon.downcase}.png"
-            when Service, ServiceTemplate
-              if item.try(:picture)
-                "/pictures/#{item.picture.basename}"
-              end
-            end
-    list_row_image(pn, image, (@listicon || view.db).underscore, item)
+    [id, item]
+  end
+  private :listicon_get_id_item
+
+  # Return the icon classname for the list view icon of a db,id pair
+  # this always supersedes listicon_image if not nil
+  def listicon_icon(view, id = nil)
+    id, item = listicon_get_id_item(view, id)
+
+    case item
+    when ExtManagementSystem
+      item.fonticon
+    else
+      nil
+    end
+  end
+
+  # Return the image name for the list view icon of a db,id pair
+  def listicon_image(view, id = nil)
+    id, item = listicon_get_id_item(view, id)
+
+    path = case item
+           when ExtManagementSystem
+             {:dir  => 'icons/new/',
+              :file => "vendor-#{item.image_name}",
```

Comment:
This is what I was going for, btw, with what I was talking about with using the ems_type (note that it doesn't *have* to be ems_type, but I'm thinking a short token that represents the item in question but isn't literally an image name or css class name).